middlewares/couponCodeGenerator.js

```diff
@@ -1,13 +1,62 @@
 /**
  * Created by ekinr on 2016/11/7.
  */
+
+var showError = function (status, err, next){
+  err.api = true;
+  err.status = status;
+  next(err);
+}
 exports.useMobileAsCode = function (req, res, next) {
   var code = req.body.mobile;
-  if(code) {
+  var username = req.params.username;
 
+  if(code) {
     // Only allow valid China mobile number
     if(/1[34578]\d{9}$/.test(code)) {
       req.couponCode = code;
+      req.userName = username;
+
+      next();
+    }
+    else {
+      var err = new Error('Invalid Mobile Provided');
+      showError(406, err, next);
+
+    }
+   
+  }
+  else {
+    var err = new Error('No Mobile Provided');
+    showError(406, err, next);
+  }
+};
+
+exports.useMobileAndUsernameToCreateCouponCode = function (req, res, next) {
```

Comment:
coupon generator should only host the algorithm that generates the actual coupon code.